rpcserver.go

```diff
@@ -1283,14 +1283,30 @@ func (r *rpcServer) EstimateFee(ctx context.Context,
 		return nil, err
 	}
 
+	var selectOutpoints fn.Set[wire.OutPoint]
+	if len(in.Inputs) != 0 {
+		wireOutpoints, err := toWireOutpoints(in.Inputs)
+		if err != nil {
+			return nil, fmt.Errorf("can't create outpoints "+
+				"%w", err)
+		}
+
+		if fn.HasDuplicates(wireOutpoints) {
+			return nil, fmt.Errorf("selected outpoints contain " +
+				"duplicate values")
+		}
+
+		selectOutpoints = fn.NewSet(wireOutpoints...)
+	}
+
 	// We will ask the wallet to create a tx using this fee rate. We set
 	// dryRun=true to avoid inflating the change addresses in the db.
 	var tx *txauthor.AuthoredTx
 	wallet := r.server.cc.Wallet
 	err = wallet.WithCoinSelectLock(func() error {
 		tx, err = wallet.CreateSimpleTx(
-			nil, outputs, feePerKw, minConfs, coinSelectionStrategy,
-			true,
+			selectOutpoints, outputs, feePerKw, minConfs,
+			coinSelectionStrategy, true,
```

Comment:
does the coinselection strategy still have an effedt if we select inputs ?

Reply:
The `selectOutpoints` end up as `inputs` in `CreateSimpleTx`: https://github.com/lightningnetwork/lnd/blob/9a83b3838f2339f4d68285fc9b1d1c5e55d52d09/lnwallet/btcwallet/btcwallet.go#L951
The strategy is then ignored in the wallet code if `selectOutpoints` is set, see: https://github.com/btcsuite/btcwallet/blob/0418e81774bdca20c5b34172e64ab26735de5008/wallet/createtx.go#L194